### PR TITLE
[Feature] Warning helper

### PR DESCRIPTION
**Pull request by antoine4livre**
## Added

- **Base:**
  - Add the `$warn()` helper ([1071c75](https://github.com/studiometa/js-toolkit/commit/1071c75))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,9 @@ All notable changes to this project will be documented in this file. The format
 
 ### Added
 
-- **Base:** add support for multiple component declaration in data-component attributes ([#330](https://github.com/studiometa/js-toolkit/pull/330))
+- **Base:**
+  - Add support for multiple component declaration in data-component attributes ([#330](https://github.com/studiometa/js-toolkit/pull/330))
+  - Add the `$warn()` helper ([1071c75](https://github.com/studiometa/js-toolkit/commit/1071c75))
```
Comment (titouanmathis) [inline CHANGELOG.md:11]: **nitpick:** Same as in studiometa/ui, prefer a reference to a PR when possible

```suggestion
  - Add the `$warn()` helper ([#335](https://github.com/studiometa/js-toolkit/pull/335))
```

File: packages/docs/api/instance-methods.md [hunk 2]
```diff
@@ -25,6 +25,31 @@ export default class Component extends Base {
...
+import { Base } from '@studiometa/js-toolkit';
+
+export default class Component extends Base {
+  static config = {
+    name: 'Component',
+    log: true,
+  };
+
+  mounted() {
+    this.$warn('Warning !');
```
Comment (titouanmathis) [inline packages/docs/api/instance-methods.md:48]: **suggestion:** Add a comment showing what will be logged to the console 

```suggestion
    this.$warn('Warning!'); // [Component-1] Warning!
```

File: packages/js-toolkit/Base/index.ts [hunk 3]
```diff
@@ -232,6 +232,13 @@ export class Base<T extends BaseProps = BaseProps> extends EventTarget {
     return this.__options.log ? window.console.log.bind(window, `[${this.__config.name}]`) : noop;
   }
 
+  /**
+   * Small helper to make warning easier.
+   */
+  get $warn(): (...args: unknown[]) => void {
```
Comment (titouanmathis) [inline packages/js-toolkit/Base/index.ts:238]: **issue (blocking):** Add tests to ensure the implementation will not break later

You can duplicate the tests for the `this.$log()` method in the `packages/tests/Base/index.spec.js` file. You will find them at the line 435 and 451:

```js
it('should have a working $log method when active', () => {
  // ...
});

it('should have a silent $log method when not active', () => {
  // ...
});
```

**Review by titouanmathis — CHANGES_REQUESTED** [review]
Great addition @antoine4livre! 👍

File: packages/js-toolkit/Base/index.ts (hunk 3 above)
Comment (antoine4livre) [reply packages/js-toolkit/Base/index.ts:238]: Done !

**Review by titouanmathis — CHANGES_REQUESTED** [review]
The unit test pipeline is failing, some snapshots are outdated following the new tests. Could your run the following command locally @antoine4livre?

```sh
npm run test -- -- -u
```

You should then have a change `*.snap` file that you can commit, this should fix the tests.

**Comment by titouanmathis** [comment]
@antoine4livre could you rebase your branch on `develop` to fix the conflicts? The `Base/index.ts` file has been renamed `Base/Base.ts`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ All notable changes to this project will be documented in this file. The format
 ### Added
 - Add a `SmartQueue` utility class ([f179d8b](https://github.com/studiometa/js-toolkit/commit/f179d8b))
 - Add a `wait` utility function ([7c9d038](https://github.com/studiometa/js-toolkit/commit/7c9d038))
+- Add the `$warn()` helper ([#335](https://github.com/studiometa/js-toolkit/pull/335))
```
Comment (titouanmathis) [inline CHANGELOG.md:10]: **nitpick:** Add a `**Base:**` prefix to the changelog entry for easier discovery of new things

```suggestion
- **Base:** add a `$warn()` helper ([#335](https://github.com/studiometa/js-toolkit/pull/335))
```

**Review by titouanmathis — APPROVED** [review]
👍

**Pull request merged**